src/languages/sqf.js

```diff
@@ -1,2529 +1,2630 @@
+    'vehicle',
+    'vehicleCargoEnabled',
+    'vehicleChat',
+    'vehicleMoveInfo',
+    'vehicleRadio',
+    'vehicleReceiveRemoteTargets',
+    'vehicleReportOwnPosition',
+    'vehicleReportRemoteTargets',
+    'vehicles',
+    'vehicleVarName',
+    'velocity',
+    'velocityModelSpace',
+    'verifySignature',
+    'vest',
+    'vestContainer',
+    'vestItems',
+    'vestMagazines',
+    'viewDistance',
+    'visibleCompass',
+    'visibleGps',
+    'visibleMap',
+    'visiblePosition',
+    'visiblePositionASL',
+    'visibleScoretable',
+    'visibleWatch',
+    'waves',
+    'waypointAttachedObject',
+    'waypointAttachedVehicle',
+    'waypointAttachObject',
+    'waypointAttachVehicle',
+    'waypointBehaviour',
+    'waypointCombatMode',
+    'waypointCompletionRadius',
+    'waypointDescription',
+    'waypointForceBehaviour',
+    'waypointFormation',
+    'waypointHousePosition',
+    'waypointLoiterAltitude',
+    'waypointLoiterRadius',
+    'waypointLoiterType',
+    'waypointName',
+    'waypointPosition',
+    'waypoints',
+    'waypointScript',
+    'waypointsEnabledUAV',
+    'waypointShow',
+    'waypointSpeed',
+    'waypointStatements',
+    'waypointTimeout',
+    'waypointTimeoutCurrent',
+    'waypointType',
+    'waypointVisible',
+    'weaponAccessories',
+    'weaponAccessoriesCargo',
+    'weaponCargo',
+    'weaponDirection',
+    'weaponInertia',
+    'weaponLowered',
+    'weaponReloadingTime',
+    'weapons',
+    'weaponsInfo',
+    'weaponsItems',
+    'weaponsItemsCargo',
+    'weaponState',
+    'weaponsTurret',
+    'weightRTD',
+    'WFSideText',
+    'wind',
+    'windDir',
+    'windRTD',
+    'windStr',
+    'wingsForcesRTD',
+    'worldName',
+    'worldSize',
+    'worldToModel',
+    'worldToModelVisual',
+    'worldToScreen'
+    ];
+
+  // list of keywords from:
+  // https://community.bistudio.com/wiki/PreProcessor_Commands
+  const PREPROCESSOR = {
+    className: 'meta',
+    begin: /#\s*[a-z]+\b/,
+    end: /$/,
+    keywords: {
+      keyword:
+        'define undef ifdef ifndef else endif include'
+    },
+    contains: [
+      {
+        begin: /\\\n/,
+        relevance: 0
+      },
+      hljs.inherit(STRINGS, {
+        className: 'string'
+      }),
+      {
+        className: 'string',
+        begin: /<[^\n>]*>/,
+        end: /$/,
+        illegal: '\\n'
+      },
+      hljs.C_LINE_COMMENT_MODE,
+      hljs.C_BLOCK_COMMENT_MODE
+    ]
+  };
+
+  return {
+    name: 'SQF',
+    case_insensitive: true,
+    keywords: {
+      keyword: KEYWORDS,
+      built_in: BUILT_IN,
+      literal: LITERAL
+    },
+    contains: [
+      hljs.C_LINE_COMMENT_MODE,
+      hljs.C_BLOCK_COMMENT_MODE,
+      hljs.NUMBER_MODE,
+      VARIABLE,
+      FUNCTION,
+      STRINGS,
+      PREPROCESSOR
+    ],
+      illegal: [
+      //$ is only valid when used with Hex numbers (e.g. $FF) or at the beginning of localized strings (e.g. "$STR_HELLO")
+      /\$[^Sa-fA-F0-9]/, 

```

Comment:
What is the `S` here?

Reply:
> What is the `S` here?

I wasn't sure if it would flag a string like `"$STR_bla"` as illegal. (which are localized strings)

Reply:
I'm not sure I yet follow how one knows it's a localized string - they alway have a $STR prefix?  If it's actually quoted though then it's already handled by STRINGS rule and doesn't need to be guarded against here since its inside the `string` scope already, unless I'm missing something?

Reply:
> I'm not sure I yet follow how one knows it's a localized string - they alway have a $STR prefix? If it's actually quoted though then it's already handled by STRINGS rule and doesn't need to be guarded against here since its inside the `string` scope already, unless I'm missing something?

Yeah if the string rule prevents it I should remove it. I didn't actually test this to be sure.